cirq-core/cirq/sim/simulator_base_test.py

```diff
@@ -192,3 +256,81 @@ def test_run_non_terminal_measurement():
     sim = CountingSimulator()
     r = sim.run(cirq.Circuit(cirq.X(q0), cirq.measure(q0), cirq.X(q0)), repetitions=2)
     assert np.allclose(r.measurements['0'], [[1], [1]])
+
+
+def test_integer_initial_state_is_split():
+    sim = SplittableCountingSimulator()
+    args = sim._create_act_on_args(2, (q0, q1))
+    assert len(set(args.values())) == 3
+    assert args[q0].state == 1
+    assert args[q1].state == 0
+    assert args[None].state == 0
+
+
+def test_integer_initial_state_is_not_split_if_disabled():
+    sim = SplittableCountingSimulator(split_untangled_states=False)
+    args = sim._create_act_on_args(2, (q0, q1))
+    assert isinstance(args, SplittableCountingActOnArgs)
+    assert args.state == 2
```

Comment:
Looking at the difference in behavior here, it seems like `ActOnArgs` should support `__getitem__` (returning `args.state` for qubits in the state, and an error otherwise) for parity with `ActOnArgsContainer`.